task/backend.go

```diff
@@ -0,0 +1,10 @@
+package task
+
+// Backend represents some storage solution to persist task objects.
+type Backend interface {
```

Comment:
I would suggest naming this `Storage` or `Repository` then. `Backend` is a bit too generic IMO.